frame/plasm-lockdrop/src/lib.rs

```diff
@@ -549,13 +588,13 @@ impl<T: Trait> Module<T> {
     fn time_bonus(duration: T::Moment) -> u16 {
```

Comment:
If this function is mutable, it may be better to express it in Trait and inject it in Runtime.

Reply:
This is constant function, probably it could be extracted from the trait

Reply:
In Mainnet Lockdrop, the
````
30 days - x24
100 days - x100
300 days - x 360
1000 days - x 1600
````
and this constant changes between Dusty and Miannet, so rewriting on merge is non-trivial The following is an example of how to make it explicitly possible to change a function by simply replacing it with the Runtime The following makes it possible for Runtime to change its functions simply by replacing them with those of the You can.


```
pub trait TimeBonusCalculator<Moment> {
  fn time_bonus(duration: Moment) -> u16;
}
```

```
trait Trait : system::Trait {
...
   type TimeBonusCalculator: TimeBonusCalculator<Self::Moment>
}
```

Reply:
@satellitex Yes, I'm totally agree